aips/move-spec-verifier.md

```diff
@@ -0,0 +1,114 @@
+4. Literal replacement - replaces literals with other literals. For example, the `0` literal can be replaced with the `1` literal or other random literal, `true` to `false`, etc. It's possible to choose the type of the literal to be replaced. For example, it's possible to replace only boolean literals.
+5. Address replacement - replaces addresses with other addresses. For example, the `0x1`/`@std` address can be replaced with the `0x000A` address or another random address.
+6. Return value replacement - replaces return values with other return values. For example, the concrete expressions can be replaced with concrete literals or other random literals.
+7. Break/continue replacement or deletion - replaces or deletes break/continue statements with other break/continue statements.
+
+The `spec-verify` command is responsible for running the Move mutator tool and then checking if the generated mutants are valid according to the Move specifications. It will do the following:
+1. Take command line arguments both for the Move Prover tool and for the Move mutator tool. It can also read the configuration from the JSON configuration file.
+2. Run the Move mutator tool (`mutate`) to generate mutants with the previously specified parameters.
+3. Run the Move Prover tool, passing the generated mutants individually.
+4. Collect the results and generate a report.
+
+The report contains information about the generated mutants as well as the killed ones for each tested source file. All the intermediate results are saved in the configurable (default: `mutants_output`) directory.
+
+## Reference Implementation
+
+TBD
+
+## Risks and Drawbacks
+
+There are the following risks and drawbacks of this proposal:
+- The set of mutation operators may be complete, as it is hard to predict all possible ways of introducing changes in the code. However, the tool is designed to be easily extendable, so it will be possible to add new operators in the future.
+- Verifying the Move specifications can be time-consuming, as it requires running the Move compiler (check if the mutant is valid) and then Move Prover for each mutant. The more mutation operators are enabled and the longer the file is, the more time is required to verify the specifications. There are two ways to mitigate this issue:
+ - The tool can be run in parallel, as each mutant is verified independently.
+ - The tool's parameters can be adjusted to, e.g., run only a subset of mutation operators or downsample the number of mutants.
+
+## Future Potential
+
+Parts of this project, such as the Move Mutator tool, can be used to create mutants for other purposes. For example, it can be used to create a set of mutants and check if the test cases can detect them. So, there is a significant potential to become a test suite verification tool. The main difference would be calling tests instead of the Move Prover.
```

Comment:
```suggestion
Parts of this project, such as the Move Mutator tool, can be used to create mutants for other purposes. For example, it can be used to create a set of mutants and check if the test cases can detect them. So, there is a significant potential to become a mutation testing tool. The main difference would be calling tests instead of the Move Prover.
```

Reply:
Corrected.